### ISSUE TITLE
how to send a BINARY through socketio?

### ISSUE THREAD

**Issue by googidaddy** [Links2004/arduinoWebSockets#716]
like ArduinoWebSockets, sendBinary(const char* a, len)?

**Comment by googidaddy** [comment]
mainly using in  esp32-cam, send image binary data

**Comment by NiklasProbst** [comment]
I don't know how to do it either. I can't find any description of when to use which type of send (.send(), .sendBIN(), .sendEVENT(), ...)

**Comment by Links2004** [comment]
`sendBIN` is used to send binary data (the other side needs to be able to handle binary)

https://github.com/Links2004/arduinoWebSockets/blob/5881a99a300672a7f6de61de76ba68f10a69ae96/src/WebSocketsClient.h#L87-L88

**Comment by sencin** [comment]
It disconnect when i try to send binary using sendBIN() using socketio protocol

**Comment by dtbaker** [comment]
Has anybody figured out how to send binary data using socket io from ESP32? 

The `sendBIN` and `sendTXT` are from normal websockets, not the socketio client.

**Comment by dtbaker** [comment]
have a look over in this thread, I was able to get binary working based on this example.

https://github.com/Links2004/arduinoWebSockets/issues/603#issuecomment-1153102225

if it continues to work well, I might open a PR to this library and introduce a sendBIN method for socket io

**Comment by sencin** [comment]
That's a custom made SendBIN right?

**Comment by dtbaker** [comment]
yes it's a custom sendBIN that sends the expected socketio message before the actual binary data.

I ran both a socketio and plain websocket server on my local computer, and ran both connections at the same time from my esp32. Each loop() would capture a frame from the camera and send it to socketio AND websocket. Debug output around each "send" call showed that the timing of both socketio and plain websocket was similar. 


I think a plain websocket connection would be simplier for binary, but I'm trying to integrate with an existing platform that is built on socketio so I'm going to stick with it
